Migrating from 1.21.11 to 26.1.1 unsuccessful

### Stack trace

```
https://mclo.gs/QpzaEqQ
```


### Plugin and Datapack List

EssentialsX
EssentialsXAntiBuild
EssentialsXDiscord
GSit
LuckPerms
SimpleVoiceChat

<img width="459" height="333" alt="Image" src="https://github.com/user-attachments/assets/eadf3c99-cf1f-4fbc-a7ff-fb5bf84d33e1" />

### Actions to reproduce (if known)

Run the Paper 26.1.1 build #18 .jar on a 1.21.11 world

### Paper version

Unable to access.

### Other

_No response_

[04:54:19] [ServerMain/INFO]: [bootstrap] Running Java 25 (OpenJDK 64-Bit Server VM 25.0.2+10-LTS; Eclipse Adoptium Temurin-25.0.2+10) on Linux 6.8.0-107-generic (amd64)
[04:54:19] [ServerMain/INFO]: [bootstrap] Loading Paper 26.1.1-18-dev/26.1@3d1da60 (2026-04-05T00:18:12Z) for Minecraft 26.1.1
[04:54:20] [ServerMain/INFO]: [PluginInitializerManager] Initializing plugins...
[04:54:21] [ServerMain/INFO]: [PluginInitializerManager] Initialized 6 plugins
[04:54:21] [ServerMain/INFO]: [PluginInitializerManager] Bukkit plugins (6):
 - Essentials (2.22.0-dev+83-08b7fe3), EssentialsAntiBuild (2.22.0-dev+83-08b7fe3), EssentialsDiscord (2.22.0-dev+83-08b7fe3), GSit (3.3.1), LuckPerms (5.5.17), voicechat (2.6.15)
[04:54:29] [ServerMain/INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, profilesHost=https://api.mojang.com, name=PROD]
[04:54:29] [ServerMain/INFO]: Starting upgrade for world "world"
[04:54:30] [ServerMain/ERROR]: Failed to start the minecraft server
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot read field "unsupportedSettings" because the return value of "io.papermc.paper.configuration.GlobalConfiguration.get()" is null
	at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:323) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:376) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:695) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:683) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2227) ~[?:?]
	at net.minecraft.util.filefix.access.ChunkNbt.updateChunk(ChunkNbt.java:26) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.fixes.LegacyStructureFileFix.storeLegacyStructureDataToChunks(LegacyStructureFileFix.java:212) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.fixes.LegacyStructureFileFix.extractAndStoreLegacyStructureData(LegacyStructureFileFix.java:149) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.fixes.LegacyStructureFileFix.lambda$makeFixer$4(LegacyStructureFileFix.java:93) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.operations.ModifyContent.lambda$fix$0(ModifyContent.java:23) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at java.base/jdk.internal.vm.ScopedValueContainer.runWithoutScope(ScopedValueContainer.java:112) ~[?:?]
	at java.base/jdk.internal.vm.ScopedValueContainer.run(ScopedValueContainer.java:98) ~[?:?]
	at java.base/java.lang.ScopedValue$Carrier.run(ScopedValue.java:466) ~[?:?]
	at net.minecraft.util.filefix.operations.ModifyContent.fix(ModifyContent.java:21) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.FileFix.runFixOperations(FileFix.java:52) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.FileFixerUpper.applyFileFixers(FileFixerUpper.java:218) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.FileFixerUpper.applyFileFixersOnCow(FileFixerUpper.java:197) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.FileFixerUpper.startOrContinueFileFixing(FileFixerUpper.java:140) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.FileFixerUpper.fix(FileFixerUpper.java:92) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.FileFixerUpper.fix(FileFixerUpper.java:76) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.server.Main.main(Main.java:207) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at io.papermc.paper.PaperBootstrap.boot(PaperBootstrap.java:21) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at org.bukkit.craftbukkit.Main.main(Main.java:234) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at io.papermc.paperclip.Paperclip.lambda$main$0(Paperclip.java:42) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:1474) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot read field "unsupportedSettings" because the return value of "io.papermc.paper.configuration.GlobalConfiguration.get()" is null
	at net.minecraft.world.level.chunk.storage.RegionFileVersion.getCompressionFormat(RegionFileVersion.java:54) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.world.level.chunk.storage.RegionFile.<init>(RegionFile.java:423) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.world.level.chunk.storage.RegionFileStorage.moonrise$getRegionFileIfExists(RegionFileStorage.java:127) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.world.level.chunk.storage.RegionFileStorage.getRegionFile(RegionFileStorage.java:279) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.world.level.chunk.storage.RegionFileStorage.read(RegionFileStorage.java:349) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.world.level.chunk.storage.SimpleRegionStorage.read(SimpleRegionStorage.java:50) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	at net.minecraft.util.filefix.access.ChunkNbt.updateChunk(ChunkNbt.java:25) ~[paper-26.1.1.jar:26.1.1-18-3d1da60]
	... 19 more

@Doc94 

Link to the world files + `server.properties` https://drive.google.com/file/d/1aZ7jjT4f4PRnsP8pqT-GUL9AdCaqUXJY/view?usp=sharing

Tried with `paper-26.1.1-20.jar` and `paper-26.1.1-24.jar` on both Windows 10 and Pop_OS 22.04 (Ubuntu based)

Both produce the same NullPointerException for "unsupportedSettings" as detailed below:

```
[20:14:29 INFO]: [bootstrap] Running Java 25 (OpenJDK 64-Bit Server VM 25.0.2+10-Ubuntu-122.04; Ubuntu null) on Linux 6.17.9-76061709-generic (amd64)
[20:14:29 INFO]: [bootstrap] Loading Paper 26.1.1-24-dev/26.1@c53ac8a (2026-04-05T18:50:48Z) for Minecraft 26.1.1
[20:14:30 INFO]: [PluginInitializerManager] Initializing plugins...
[20:14:30 INFO]: [PluginInitializerManager] Initialized 0 plugins
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by org.joml.MemUtil$MemUtilUnsafe (file:/pool/media/backups/26.1-min/libraries/org/joml/joml/1.10.8/joml-1.10.8.jar)
WARNING: Please consider reporting this to the maintainers of class org.joml.MemUtil$MemUtilUnsafe
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release
[20:14:34 INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, profilesHost=https://api.mojang.com, name=PROD]
[20:14:35 INFO]: Starting upgrade for world "hub"
[20:14:35 ERROR]: Failed to start the minecraft server
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot read field "unsupportedSettings" because the return value of "io.papermc.paper.configuration.GlobalConfiguration.get()" is null
        at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:323) ~[?:?]
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:376) ~[?:?]
        at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:695) ~[?:?]
        at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:683) ~[?:?]
        at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2227) ~[?:?]
        at net.minecraft.util.filefix.access.ChunkNbt.updateChunk(ChunkNbt.java:26) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.fixes.LegacyStructureFileFix.storeLegacyStructureDataToChunks(LegacyStructureFileFix.java:212) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.fixes.LegacyStructureFileFix.extractAndStoreLegacyStructureData(LegacyStructureFileFix.java:149) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.fixes.LegacyStructureFileFix.lambda$makeFixer$4(LegacyStructureFileFix.java:93) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.operations.ModifyContent.lambda$fix$0(ModifyContent.java:23) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at java.base/jdk.internal.vm.ScopedValueContainer.runWithoutScope(ScopedValueContainer.java:112) ~[?:?]
        at java.base/jdk.internal.vm.ScopedValueContainer.run(ScopedValueContainer.java:98) ~[?:?]
        at java.base/java.lang.ScopedValue$Carrier.run(ScopedValue.java:466) ~[?:?]
        at net.minecraft.util.filefix.operations.ModifyContent.fix(ModifyContent.java:21) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.FileFix.runFixOperations(FileFix.java:52) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.FileFixerUpper.applyFileFixers(FileFixerUpper.java:218) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.FileFixerUpper.applyFileFixersOnCow(FileFixerUpper.java:197) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.FileFixerUpper.startOrContinueFileFixing(FileFixerUpper.java:140) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.FileFixerUpper.fix(FileFixerUpper.java:92) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.FileFixerUpper.fix(FileFixerUpper.java:76) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.server.Main.main(Main.java:207) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at io.papermc.paper.PaperBootstrap.boot(PaperBootstrap.java:21) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at org.bukkit.craftbukkit.Main.main(Main.java:234) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at io.papermc.paperclip.Paperclip.lambda$main$0(Paperclip.java:42) ~[?:?]
        at java.base/java.lang.Thread.run(Thread.java:1474) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot read field "unsupportedSettings" because the return value of "io.papermc.paper.configuration.GlobalConfiguration.get()" is null
        at net.minecraft.world.level.chunk.storage.RegionFileVersion.getCompressionFormat(RegionFileVersion.java:54) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.world.level.chunk.storage.RegionFile.<init>(RegionFile.java:423) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.world.level.chunk.storage.RegionFileStorage.moonrise$getRegionFileIfExists(RegionFileStorage.java:127) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.world.level.chunk.storage.RegionFileStorage.getRegionFile(RegionFileStorage.java:279) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.world.level.chunk.storage.RegionFileStorage.read(RegionFileStorage.java:349) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.world.level.chunk.storage.SimpleRegionStorage.read(SimpleRegionStorage.java:50) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        at net.minecraft.util.filefix.access.ChunkNbt.updateChunk(ChunkNbt.java:25) ~[paper-26.1.1.jar:26.1.1-24-c53ac8a]
        ... 19 more
```

**edit:** From discussion with server staff, we believe this world may have originally been created in 1.12 / 1.13 and has been running since as our creative flat world / hub to other servers.

--- a/paper-server/patches/features/0001-Moonrise-optimisation-patches.patch
+++ b/paper-server/patches/features/0001-Moonrise-optimisation-patches.patch
@@ -33593,7 +33593,7 @@ index cc5d366f7b485f98901572b1aaf57d8b650d521f..7adb4be57bd9fdc9980b5b7e91e9d9ca
      private final Long2ObjectLinkedOpenHashMap<CompletableFuture<BitSet>> regionCacheForBlender = new Long2ObjectLinkedOpenHashMap<>();
      private static final int REGION_CACHE_SIZE = 1024;
 diff --git a/net/minecraft/world/level/chunk/storage/RegionFile.java b/net/minecraft/world/level/chunk/storage/RegionFile.java
-index 728ec122b7af090427cc7511a168336d539835a1..5149635679ec03d881e63effbd79fc193d7e52c8 100644
+index 9409cc5c8dbd12ce6bdf34843da70170ebce41bd..c9acbdce97cb605a45124bb9d3944070396adc03 100644
 --- a/net/minecraft/world/level/chunk/storage/RegionFile.java
 +++ b/net/minecraft/world/level/chunk/storage/RegionFile.java
 @@ -23,7 +23,7 @@ import net.minecraft.world.level.ChunkPos;
@@ -33625,7 +33625,7 @@ index 728ec122b7af090427cc7511a168336d539835a1..5149635679ec03d881e63effbd79fc19
 +    // Paper end - rewrite chunk system
 +
      public RegionFile(final RegionStorageInfo info, final Path path, final Path externalFileDir, final boolean sync) throws IOException {
-         this(info, path, externalFileDir, RegionFileVersion.getCompressionFormat(), sync); // Paper - Configurable region compression format
+         this(info, path, externalFileDir, RegionFileVersion.getSelected(), sync);
      }
 @@ -201,6 +216,15 @@ public class RegionFile implements AutoCloseable {
      }

--- a/paper-server/patches/features/0019-Attempt-to-recalculate-regionfile-header-if-it-is-co.patch
+++ b/paper-server/patches/features/0019-Attempt-to-recalculate-regionfile-header-if-it-is-co.patch
@@ -55,7 +55,7 @@ index a6f9435a8e2a6c8cc05bd3873cb6fc0ec98973b8..d44fd9eea0ccb09c894d6431ede8a4c2
          this.used.set(position, position + size);
      }
 diff --git a/net/minecraft/world/level/chunk/storage/RegionFile.java b/net/minecraft/world/level/chunk/storage/RegionFile.java
-index fc712f9dbc893a6dee8ccecc54d0af07659dee62..b4a59b855b45018853e8122a5829a48ebd220a5f 100644
+index 5e77906d77da82b23fbd6e8d16fbd0847c83f4da..e077797cefffcb9db627cab12c7d88245e5baa69 100644
 --- a/net/minecraft/world/level/chunk/storage/RegionFile.java
 +++ b/net/minecraft/world/level/chunk/storage/RegionFile.java
 @@ -47,6 +47,363 @@ public class RegionFile implements AutoCloseable, ca.spottedleaf.moonrise.patche
@@ -760,7 +760,7 @@ index 5fa64ff30b8252f139ca7f77c14352c384c8c0e5..eae4963b6d18c5f92871477b93f81c2c
  
          return var4;
 diff --git a/net/minecraft/world/level/chunk/storage/RegionFileVersion.java b/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
-index 97d3da90c270a06d85e70d261658f71d97ae4e3c..11b65c688144e7ef5b7cba603949843a11ccd084 100644
+index ea229f63f81c5638ded74c28e1e13d8948f2e5e0..d991fccfc3d2cc51611f802a6d40429024e21d9a 100644
 --- a/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
 +++ b/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
 @@ -21,7 +21,7 @@ import org.slf4j.Logger;
@@ -771,7 +771,7 @@ index 97d3da90c270a06d85e70d261658f71d97ae4e3c..11b65c688144e7ef5b7cba603949843a
 +    public static final Int2ObjectMap<RegionFileVersion> VERSIONS = new Int2ObjectOpenHashMap<>(); // Paper - private -> public
      private static final Object2ObjectMap<String, RegionFileVersion> VERSIONS_BY_NAME = new Object2ObjectOpenHashMap<>();
      public static final RegionFileVersion VERSION_GZIP = register(
-         new RegionFileVersion(1, null, in -> new FastBufferedInputStream(new GZIPInputStream(in)), out -> new BufferedOutputStream(new GZIPOutputStream(out)))
+         new RegionFileVersion(1, "gzip", in -> new FastBufferedInputStream(new GZIPInputStream(in)), out -> new BufferedOutputStream(new GZIPOutputStream(out))) // Paper - Enable for Configurable region compression format
 diff --git a/net/minecraft/world/level/chunk/storage/RegionStorageInfo.java b/net/minecraft/world/level/chunk/storage/RegionStorageInfo.java
 index 590bd1ca60956e5c5be58481f97401e55f9e1aeb..4e0c9fa1eecfa9f21285b7911e25830bbcbc8d86 100644
 --- a/net/minecraft/world/level/chunk/storage/RegionStorageInfo.java
@@ -791,7 +791,7 @@ index 590bd1ca60956e5c5be58481f97401e55f9e1aeb..4e0c9fa1eecfa9f21285b7911e25830b
          return new RegionStorageInfo(this.level, this.dimension, this.type + suffix);
      }
 diff --git a/net/minecraft/world/level/chunk/storage/SerializableChunkData.java b/net/minecraft/world/level/chunk/storage/SerializableChunkData.java
-index 22a83f312d357176a5489e5b6b71c3ef3e737df7..74158a5d7ea8058c99ab215265987746aa043fc7 100644
+index 1e0afa9590de7fe492cf29aded96be3db498c51e..df0bb06176ad5cce3ed8785d22d68b2dcbf5a8ca 100644
 --- a/net/minecraft/world/level/chunk/storage/SerializableChunkData.java
 +++ b/net/minecraft/world/level/chunk/storage/SerializableChunkData.java
 @@ -120,6 +120,18 @@ public record SerializableChunkData(
@@ -823,7 +823,7 @@ index 22a83f312d357176a5489e5b6b71c3ef3e737df7..74158a5d7ea8058c99ab215265987746
          tag.putString("Status", BuiltInRegistries.CHUNK_STATUS.getKey(this.chunkStatus).toString());
          tag.storeNullable("blending_data", BlendingData.Packed.CODEC, this.blendingData);
 diff --git a/net/minecraft/world/level/chunk/storage/SimpleRegionStorage.java b/net/minecraft/world/level/chunk/storage/SimpleRegionStorage.java
-index 884fa9545a93008fa92863c845aef8f18d88e25a..15da3686bb1b5fddd388a88a7b300aa221c83f58 100644
+index 6e55cb304fd9a5a01bbe7666dddb882e684cdcf1..6d954064f409ddd2cb7cfd0d0c428ac186f5ca4d 100644
 --- a/net/minecraft/world/level/chunk/storage/SimpleRegionStorage.java
 +++ b/net/minecraft/world/level/chunk/storage/SimpleRegionStorage.java
 @@ -34,6 +34,7 @@ public class SimpleRegionStorage implements ca.spottedleaf.moonrise.patches.chun

--- a/paper-server/patches/sources/net/minecraft/world/level/chunk/storage/RegionFile.java.patch
+++ b/paper-server/patches/sources/net/minecraft/world/level/chunk/storage/RegionFile.java.patch
@@ -1,14 +1,5 @@
 --- a/net/minecraft/world/level/chunk/storage/RegionFile.java
 +++ b/net/minecraft/world/level/chunk/storage/RegionFile.java
-@@ -47,7 +_,7 @@
-     protected final RegionBitmap usedSectors = new RegionBitmap();
- 
-     public RegionFile(final RegionStorageInfo info, final Path path, final Path externalFileDir, final boolean sync) throws IOException {
--        this(info, path, externalFileDir, RegionFileVersion.getSelected(), sync);
-+        this(info, path, externalFileDir, RegionFileVersion.getCompressionFormat(), sync); // Paper - Configurable region compression format
-     }
- 
-     public RegionFile(final RegionStorageInfo info, final Path path, final Path externalFileDir, final RegionFileVersion version, final boolean sync) throws IOException {
 @@ -83,6 +_,14 @@
                      if (offset != 0) {
                          int sectorNumber = getSectorNumber(offset);

--- a/paper-server/patches/sources/net/minecraft/world/level/chunk/storage/RegionFileVersion.java.patch
+++ b/paper-server/patches/sources/net/minecraft/world/level/chunk/storage/RegionFileVersion.java.patch
@@ -1,19 +1,11 @@
 --- a/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
 +++ b/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
-@@ -49,6 +_,16 @@
-     private final RegionFileVersion.StreamWrapper<InputStream> inputWrapper;
-     private final RegionFileVersion.StreamWrapper<OutputStream> outputWrapper;
- 
-+    // Paper start - Configurable region compression format
-+    public static RegionFileVersion getCompressionFormat() {
-+        return switch (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.compressionFormat) {
-+            case GZIP -> VERSION_GZIP;
-+            case ZLIB -> VERSION_DEFLATE;
-+            case LZ4 -> VERSION_LZ4;
-+            case NONE -> VERSION_NONE;
-+        };
-+    }
-+    // Paper end - Configurable region compression format
-     private RegionFileVersion(
-         final int id,
-         final @Nullable String optionName,
+@@ -24,7 +_,7 @@
+     private static final Int2ObjectMap<RegionFileVersion> VERSIONS = new Int2ObjectOpenHashMap<>();
+     private static final Object2ObjectMap<String, RegionFileVersion> VERSIONS_BY_NAME = new Object2ObjectOpenHashMap<>();
+     public static final RegionFileVersion VERSION_GZIP = register(
+-        new RegionFileVersion(1, null, in -> new FastBufferedInputStream(new GZIPInputStream(in)), out -> new BufferedOutputStream(new GZIPOutputStream(out)))
++        new RegionFileVersion(1, "gzip", in -> new FastBufferedInputStream(new GZIPInputStream(in)), out -> new BufferedOutputStream(new GZIPOutputStream(out))) // Paper - Enable for Configurable region compression format
+     );
+     public static final RegionFileVersion VERSION_DEFLATE = register(
+         new RegionFileVersion(

--- a/paper-server/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ b/paper-server/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
@@ -183,19 +183,10 @@ public class GlobalConfiguration extends ConfigurationPart {
         public boolean allowHeadlessPistons = false;
         @Comment("This setting controls if the vanilla damage tick should be skipped if damage was blocked via a shield.")
         public boolean skipVanillaDamageTickWhenShieldBlocked = false;
-        @Comment("This setting controls what compression format is used for region files.")
-        public CompressionFormat compressionFormat = CompressionFormat.ZLIB;
         @Comment("This setting controls if equipment should be updated when handling certain player actions.")
         public boolean updateEquipmentOnPlayerActions = true;
         @Comment("This setting controls what item data components don't need to be sanitized in oversized item obfuscation. Adding them re-enables exploits, but may be needed for certain resource packs. (Expected: minecraft:container, minecraft:charged_projectiles and minecraft:bundle_contents)")
         public OversizedItemComponentSanitizer.AssetOversizedItemComponentSanitizerConfiguration oversizedItemComponentSanitizer = new OversizedItemComponentSanitizer.AssetOversizedItemComponentSanitizerConfiguration(Set.of());
-
-        public enum CompressionFormat {
-            GZIP,
-            ZLIB,
-            LZ4,
-            NONE
-        }
     }
 
     public Commands commands;

--- a/paper-server/src/main/java/io/papermc/paper/configuration/RemovedConfigurations.java
+++ b/paper-server/src/main/java/io/papermc/paper/configuration/RemovedConfigurations.java
@@ -86,6 +86,7 @@ interface RemovedConfigurations {
         path("misc", "fix-entity-position-desync"),
         path("chunk-system", "gen-parallelism"),
         path("logging"), // server is no longer obfuscated since 26.1
+        path("unsupported-settings", "compression-format"),
     };
 
 }

